### PR TITLE
print out extra files in data dir instead of erroring

### LINKED ISSUE #2571: etcd fails to bootstrap in an non-empty data dir
```
$ tree -a 1.etcd
1.etcd
└── .aaa
$ ./bin/etcd --data-dir 1.etcd
2015/03/24 11:29:20 etcd: found invalid file/dir .aaa under data dir 1.etcd (Ignore this if you are upgrading etcd)
2015/03/24 11:29:20 etcd: listening for peers on http://localhost:2380
2015/03/24 11:29:20 etcd: listening for peers on http://localhost:7001
2015/03/24 11:29:20 etcd: listening for client requests on http://localhost:2379
2015/03/24 11:29:20 etcd: listening for client requests on http://localhost:4001
2015/03/24 11:29:20 etcd: stopping listening for client requests on http://localhost:4001
2015/03/24 11:29:20 etcd: stopping listening for client requests on http://localhost:2379
2015/03/24 11:29:20 etcd: stopping listening for peers on http://localhost:7001
2015/03/24 11:29:20 etcd: stopping listening for peers on http://localhost:2380
2015/03/24 11:29:20 etcd: unknown wal version in data dir 1.etcd
```

etcd failing because of an unknown file in the data dir is terrible.

/cc @kelseyhightower @crawford @barakmich

### PR DESCRIPTION
**Pull request by yichengq**
fixes #2571 
/cc @barakmich @kelseyhightower @xiang90

### CODE REVIEW

**Comment by xiang90** [comment]
can we move that out of wal? it should not be inside wal pkg.

**Comment by yichengq** [comment]
@xiang90 I need some time to do the move.

**Comment by xiang90** [comment]
LGTM. rename all walversion -> datadir version

**Comment by xiang90** [comment]
print out warning if there is unrelated files inside the data dir.

File: etcdserver/storage.go [hunk 1]
```diff
@@ -88,9 +89,9 @@ func readWAL(waldir string, snap walpb.Snapshot) (w *wal.WAL, id, cid types.ID,
 
 // upgradeWAL converts an older version of the etcdServer data to the newest version.
 // It must ensure that, after upgrading, the most recent version is present.
-func upgradeWAL(baseDataDir string, name string, ver wal.WalVersion) error {
+func upgradeWAL(baseDataDir string, name string, ver version.DataDirVersion) error {
```
Comment (xiang90) [inline etcdserver/storage.go]: upgradeDataDir?

File: etcdserver/server.go [hunk 2]
```diff
@@ -143,14 +144,28 @@ func NewServer(cfg *ServerConfig) (*EtcdServer, error) {
...
+	if dataVer == version.DataDirUnknown {
+		names, err := fileutil.ReadDir(cfg.DataDir)
+		if err != nil {
+			return nil, err
+		}
+		for _, name := range names {
+			if name == "member" || name == "proxy" {
+				continue
+			}
+			log.Printf("etcdserver: extra file %s found in data-dir %s", name, cfg.DataDir)
```
Comment (xiang90) [inline etcdserver/server.go]: /cc @crawford

**Comment by yichengq** [comment]
Current behavior:
```
$ tree -la 1.etcd
1.etcd
├── .aaa
unihorn@CoreOS ~/g/s/g/c/etcd> ./bin/etcd --data-dir 1.etcd                         fix-detect-wal!?
2015/03/24 18:38:00 etcd: found invalid file/dir .aaa under data dir 1.etcd (Ignore this if you are upgrading etcd)
2015/03/24 18:38:00 etcd: listening for peers on http://localhost:2380
2015/03/24 18:38:00 etcd: listening for peers on http://localhost:7001
2015/03/24 18:38:00 etcd: listening for client requests on http://localhost:2379
2015/03/24 18:38:00 etcd: listening for client requests on http://localhost:4001
*2015/03/24 18:38:00 etcdserver: datadir is valid for the 2.0.1 format*
*2015/03/24 18:38:00 etcdserver: extra file .aaa found in data-dir 1.etcd*
2015/03/24 18:38:00 etcdserver: name = default
2015/03/24 18:38:00 etcdserver: data dir = 1.etcd
2015/03/24 18:38:00 etcdserver: member dir = 1.etcd/member
2015/03/24 18:38:00 etcdserver: heartbeat = 100ms
2015/03/24 18:38:00 etcdserver: election = 1000ms
2015/03/24 18:38:00 etcdserver: snapshot count = 10000
2015/03/24 18:38:00 etcdserver: advertise client URLs = http://localhost:2379,http://localhost:4001
2015/03/24 18:38:00 etcdserver: initial advertise peer URLs = http://localhost:2380,http://localhost:7001
2015/03/24 18:38:00 etcdserver: initial cluster = default=http://localhost:2380,default=http://localhost:7001
2015/03/24 18:38:00 etcdserver: start member ce2a822cea30bfca in cluster 7e27652122e8b2ae
2015/03/24 18:38:00 INFO: raft: ce2a822cea30bfca became follower at term 0
2015/03/24 18:38:00 INFO: raft: newRaft ce2a822cea30bfca [peers: [], term: 0, commit: 0, applied: 0, lastindex: 0, lastterm: 0]
2015/03/24 18:38:00 INFO: raft: ce2a822cea30bfca became follower at term 1
2015/03/24 18:38:00 etcdserver: added local member ce2a822cea30bfca [http://localhost:2380 http://localhost:7001] to cluster 7e27652122e8b2ae
2015/03/24 18:38:01 INFO: raft: ce2a822cea30bfca is starting a new election at term 1
2015/03/24 18:38:01 INFO: raft: ce2a822cea30bfca became candidate at term 2
2015/03/24 18:38:01 INFO: raft: ce2a822cea30bfca received vote from ce2a822cea30bfca at term 2
2015/03/24 18:38:01 INFO: raft: ce2a822cea30bfca became leader at term 2
2015/03/24 18:38:01 INFO: raft.node: ce2a822cea30bfca elected leader ce2a822cea30bfca at term 2
2015/03/24 18:38:01 etcdserver: published {Name:default ClientURLs:[http://localhost:2379 http://localhost:4001]} to cluster 7e27652122e8b2ae
```

File: etcdserver/server.go (hunk 2 above)
Comment (crawford) [inline etcdserver/server.go]: LGTM

Comment (kelseyhightower) [inline etcdserver/server.go]: Why do we need to log it? If we really must log it, we should let the user know we plan to ignore the file.

```
etcdserver: ignoring extra file extra.txt found in data-dir /var/lib/etcd
```

Comment (xiang90) [inline etcdserver/server.go]: we should suggest them to run etcd with a clean data dir. 
ignore lgtm

**Comment by kelseyhightower** [comment]
Thanks you from removing the printing of the extra files. We can educate people on the expectation of the datadir and provide docs on why a clean data dir is important, but not mandatory.

**Pull request merged**